base/linalg/diagonal.jl

```diff
@@ -135,16 +135,16 @@ function tril!(D::Diagonal,k::Integer=0)
     return D
 end
 
-==(Da::Diagonal, Db::Diagonal) = Da.diag == Db.diag
--(A::Diagonal)=Diagonal(-A.diag)
-+(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag + Db.diag)
--(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag - Db.diag)
+(==)(Da::Diagonal, Db::Diagonal) = Da.diag == Db.diag
```

Comment:
It doesn't work with any of the other function syntax, like `::T` return type annotations (`+(x::Int, y::Int)::Int`). How things are parsed is an implementation detail, sure, but it doesn't change that this syntax is incredibly brittle.